apps/studio/src/lib/db/authentication/azure.ts

```diff
@@ -153,17 +171,64 @@ export class AzureAuthService {
     }
   }
 
+  private async getAzureCLIToken(options: AzureAuthOptions): Promise<AuthConfig> {
+
+    if(!options?.cliPath){
+      throw new Error('AZ command not specified')
+    }
+
+
+    return new Promise<AuthConfig>((resolve, reject) => {
+      const proc = spawn(options?.cliPath, [
+      'account',
+      'get-access-token',
+      '--resource',
+      BksConfig.azure.azSQLLoginScope,
+      '--output',
+      'json'
+    ]);
+
+      proc.stdout.on('data', (chunk) => {
```

Comment:
@ClarLabs Several issues here with spawn:

1. `data` events can trigger multiple times, it's a buffer so it triggers again when there is more output.
2. A healthy process can emit both STDERR and STDOUT data, you can't use this to figure out if it's good or bad
3. You're not listening for the `close` event -- this is how you figure out if the process has ended (and whether it succeeded).
  ```
    ls.on('close', (code) => {
      console.log(`child process exited with code ${code}`);
    });
  ```